### ISSUE TITLE
Bugs with Player revive (1.17.1)

### ISSUE THREAD

**Issue by WardsROverrated** [ichttt/FirstAid#147]
Using both player revive and first aid for 1.17.1, latest versions of both as of today. 

[Player revive description, skip ahead if not needed]

Player revive allows you revive friends with a mechanic similar to that of a battle royal, where they will get a "bleed out" timer, after which they will die normally, if another player gets close to them and right clicks, it will start a "reviving" counter, after which the downed player will be "revived" with some health left

Link to mod in curse forge: https://www.curseforge.com/minecraft/mc-mods/playerrevive

[Bug descriptions]

There seems to be 3 bugs involving both of these mods, these bugs do not happen if only one of them is installed and 2 of them are 100% reproducible.

1) Dying with any ongoing regeneration effect will completely skip out on the bleed out timer and just kill the player, all items in the inventory will just disappear. 

I do have the error messages for this one which is the txt I provided. 

To reproduce: 
1- Acquire any level of regeneration, any duration.
2- Die while the regeneration effect is still active. 

Player will just die without the option to be revived and their items won't drop. 
We are playing in a server and the dying will player will also disconnect 

2) Dying with the poison effect and being revived by another player while the poison effect is still going will still kill the downed player and all items in the inventory will disappear.

I also have the error messages for this one, which is also in the txt.

To reproduce:
1- Acquire any level of poison, any duration.
2- Die.
3- Have another player try to revive you while the effects of poison are still active.  

Once the revive counter completes, if the downed player still has the poison effect they will disappear like if they weren't revived and items will not drop.

The dying player will not disconnect in this case. 

3) Being revived will sometimes not allow you to heal using bandages or plaster, counter for next healing will not progress, each body part will stay at 0 hp but the player will not get any debuffs. Acquiring the effects of instant health or regeneration will fix it.  

This one we could not consistently reproduce, it happened once out of 8 attempts and I couldn't find the error messages on the console. I can try to get them at a later date if you need them

To reproduce:
1- Die.
2- Be revived.
3- Try to use a bandage or plaster.

The first bug was reported on the player revive github by a different user [ https://github.com/CreativeMD/PlayerRevive/issues/68 ] And the creator said to post it here, since they don't see a way to fix it on their end so I will only post it here but if you need me I will gladly also open an issue on their end.
If there is any extra information you need I will try my best to provide it asap. 
[Bugs.txt](https://github.com/ichttt/FirstAid/files/7387260/Bugs.txt)

**Comment by n0itLT** [comment]
I am having a bug where when I set my hp back to normal after removing this mod, I get all the debuffs.. doesn't go away even death

**Comment by ichttt** [comment]
Yeah, probably my fault, I will see if I can fix this up. But I will most likely not backport this fix to 1.16 or 1.17 and only target 1.18+
